src/libraries/System.IO.Compression/src/System/IO/Compression/ZipArchiveEntry.cs

```diff
@@ -843,14 +1082,61 @@ private bool IsOpenable(bool needToUncompress, bool needToLoadIntoMemory, out st
                 {
                     return false;
                 }
-                if (!ZipLocalFileHeader.TrySkipBlock(_archive.ArchiveStream))
+
+                if (!IsEncrypted && !ZipLocalFileHeader.TrySkipBlock(_archive.ArchiveStream))
                 {
                     message = SR.LocalFileHeaderCorrupt;
                     return false;
                 }
+                else if (IsEncrypted && CompressionMethod == CompressionMethodValues.Aes)
+                {
+                    _archive.ArchiveStream.Seek(_offsetOfLocalHeader, SeekOrigin.Begin);
+                    _aesCompressionLevel = CompressionMethodValues.Aes;
+                    byte? aesStrength;
+                    ushort? originalCompressionMethod;
+                    ushort? aeVersion;
+                    uint? crc32;
+                    if (!ZipLocalFileHeader.TrySkipBlockAESAware(_archive.ArchiveStream, out aesStrength, out originalCompressionMethod, out aeVersion, out crc32))
+                    {
+                        message = SR.LocalFileHeaderCorrupt;
+                        return false;
+                    }
+
+                    if (aesStrength.HasValue)
```

Comment:
Can any of the nullable values be meaningfully absent if at least one of the others is present? Maybe the method should return `WinZipAesExtraField` type

Alternatively, what should happen if the AE extra block is missing? is that possible at this point? Maybe the variables shouldn't be nullable in the first place

Reply:
I will update the call so instead of outputting 3 different variables I will return a WinZipAesExtraField. If something is missing or is malformed I will throw an error for corrupted header. No metadata can be missing or not be set.